android/uhabits-android/src/main/java/org/isoron/uhabits/widgets/CheckmarkWidgetProvider.kt

```diff
@@ -23,7 +23,13 @@ import android.content.*
 class CheckmarkWidgetProvider : BaseWidgetProvider() {
     override fun getWidgetFromId(context: Context, id: Int): BaseWidget {
         val habits = getHabitsFromWidgetId(id)
-        if (habits.size == 1) return CheckmarkWidget(context, id, habits[0])
-        else return StackWidget(context, id, StackWidgetType.CHECKMARK, habits)
+        return if (habits.size == 1) {
+            if (habits[0].isNumerical){
+                NumericalCheckmarkWidget(context, id, habits[0])
+            }else {
+                CheckmarkWidget(context, id, habits[0])
+            }
+        }
+        else StackWidget(context, id, StackWidgetType.CHECKMARK, habits)
```

Comment:
Your changes currently break the `StackWidget`, but don't worry about this. If you implement my other suggestion about unifying `CheckmarkWidget` and `NumericalCheckmarkWidget`, this problem will be automatically solved, I think.